### PR TITLE
Gabor/tag on alloc

### PR DESCRIPTION
**Pull request by ggreif**

### CODE REVIEW

File: rts/motoko-rts/src/text.rs [hunk 1]
```diff
@@ -100,9 +100,8 @@ pub unsafe fn text_concat<M: Memory>(mem: &mut M, s1: Value, s2: Value) -> Value
     }
 
     // Create concat node
-    let r = mem.alloc_words(size_of::<Concat>());
+    let r = mem.alloc_words(size_of::<Concat>(), TAG_CONCAT);
     let r_concat = r.get_ptr() as *mut Concat;
```
Comment (osa1) [inline rts/motoko-rts/src/text.rs:104]: Now that you have the tag field initialized you could use `as_concat` here.

**Review by osa1 — APPROVED** [review]
LGTM.

File: rts/motoko-rts/src/text.rs (hunk 1 above)
Comment (ggreif) [reply rts/motoko-rts/src/text.rs:104]: I will `grep` for other such occurrences too.

Comment (ggreif) [reply rts/motoko-rts/src/text.rs]: 0cdd247!

File: rts/motoko-rts/src/gc/copying.rs [hunk 2]
```diff
@@ -144,7 +144,7 @@ unsafe fn evac<M: Memory>(
     let obj_size = object_size(obj as usize);
 
     // Allocate space in to-space for the object
-    let obj_addr = mem.alloc_words(obj_size).get_ptr();
+    let obj_addr = mem.alloc_words(obj_size, 0).get_ptr();
```
Comment (ggreif) [inline rts/motoko-rts/src/gc/copying.rs:147]: I don't like this 0

File: rts/motoko-rts/src/gc/mark_compact/mark_stack.rs [hunk 3]
```diff
@@ -41,7 +41,7 @@ pub unsafe fn free_mark_stack() {
 /// Doubles the stack size
 pub unsafe fn grow_stack<M: Memory>(mem: &mut M) {
     let stack_cap: Words<u32> = STACK_BLOB_PTR.len().to_words();
-    let p = mem.alloc_words(stack_cap).get_ptr() as *mut usize;
+    let p = mem.alloc_words(stack_cap, 0 /*FIXME: wut?*/).get_ptr() as *mut usize;
```
Comment (ggreif) [inline rts/motoko-rts/src/gc/mark_compact/mark_stack.rs:44]: same here

**Comment by ggreif** [comment]
#3334 taught me that `rustc` has pretty good `#inline` heuristics.

**Comment by luc-blaeser** [comment]
Just doing some repo cleanup for better focussing on active work:
Closing PR as it seems inactive for a longer time. Please re-open if it is still relevant and after it has been updated with latest master changes.

**Pull request closed without merging**